crypto-markets/src/exchanges/kraken_futures.rs

```diff
@@ -0,0 +1,160 @@
+use std::collections::HashMap;
+
+use super::utils::http_get;
+use crate::{
+    error::{Error, Result},
+    Fees, Market, MarketType, Precision,
+};
+
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+pub(crate) fn fetch_symbols(market_type: MarketType) -> Result<Vec<String>> {
+    match market_type {
+        MarketType::InverseFuture => fetch_inverse_future_symbols(),
+        _ => panic!("Unsupported market_type: {}", market_type),
+    }
+}
+
+pub(crate) fn fetch_markets(market_type: MarketType) -> Result<Vec<Market>> {
+    match market_type {
+        MarketType::InverseFuture => fetch_inverse_future_markets(),
+        _ => panic!("Unsupported market_type: {}", market_type),
+    }
+}
+
+#[derive(Clone, Serialize, Deserialize)]
+struct InverseFutureMarket {
+    symbol: String,
+    #[serde(alias = "type")]
+    market_type: String,
+    underlying: String,
+    #[serde(alias = "tickSize")]
+    tick_size: f64,
+    #[serde(alias = "contractSize")]
+    contract_size: f64,
+    tradeable: bool,
+    isin: String,
+    #[serde(alias = "lastTradingTime")]
+    last_trading_time: Option<String>,
+    #[serde(alias = "marginLevels")]
+    margin_levels: Vec<MarginLevel>,
+    #[serde(alias = "retailMarginLevels")]
+    retail_margin_levels: Vec<MarginLevel>,
+}
+
+#[derive(Clone, Serialize, Deserialize)]
+struct MarginLevel {
+    contracts: u64,
+    #[serde(alias = "initialMargin")]
+    initial_margin: f64,
+    #[serde(alias = "maintenanceMargin")]
+    maintenance_margin: f64,
+}
+
+#[derive(Serialize, Deserialize)]
+struct Response {
+    result: HashMap<String, InverseFutureMarket>,
+}
+
+fn check_error_in_body(resp: String) -> Result<String> {
+    let obj = serde_json::from_str::<HashMap<String, Value>>(&resp);
+    if obj.is_err() {
+        return Ok(resp);
+    }
+
+    match obj.unwrap().get("error") {
+        Some(err) => {
+            let arr = err.as_array().unwrap();
+            if arr.is_empty() {
+                Ok(resp)
+            } else {
+                Err(Error(resp))
+            }
+        }
+        None => Ok(resp),
+    }
+}
+
+pub(super) fn kraken_http_get(url: &str) -> Result<String> {
```

Comment:
Should be `kraken_futures_http_get` I believe. 
<img width="673" alt="image" src="https://user-images.githubusercontent.com/219162/154075215-a8384a7f-ab40-42a9-8754-aecb129c3ab1.png">